pkg/backend/openshiftcluster.go

```diff
@@ -130,13 +135,19 @@ func (ocb *openShiftClusterBackend) heartbeat(log *logrus.Entry, doc *api.OpenSh
 		for {
 			_, err := ocb.db.OpenShiftClusters.Lease(doc.Key)
 			if err != nil {
+				// if lost lease - cancel backend workers
+				// and revoke lease
+				if errors.Is(err, database.ErrorLostLease) {
+					cancel()
```

Comment:
I think that if we're returning, we should be cancelling regardless of the error we got?
Log that we decided to cancel.

Reply:
Current `Lease()` can return error for `unable to get()` document, `StatusPreconditionFailed`, `key` is not uppercase. We could argue that automatically some of these are not `loose lease` cases and we dont want to cancel. But on the other hand "error means failure, so defer will exit and cancel either way. :man_shrugging: